internal/pkg/mining/block_generate.go

```diff
@@ -61,53 +55,26 @@ func (w *DefaultWorker) Generate(
 
 	blockHeight := baseHeight + nullBlockCount + 1
 
-	ancestors, err := w.getAncestors(ctx, baseTipSet, types.NewBlockHeight(blockHeight))
-	if err != nil {
-		return nil, errors.Wrap(err, "get base tip set ancestors")
-	}
-
 	// Construct list of message candidates for inclusion.
 	// These messages will be processed, and those that fail excluded from the block.
 	pending := w.messageSource.Pending()
 	mq := NewMessageQueue(pending)
 	candidateMsgs := orderMessageCandidates(mq.Drain())
 
-	// run state transition to learn which messages are valid
-	vms := vm.NewStorageMap(w.blockstore)
-	results, err := w.processor.ApplyMessagesAndPayRewards(ctx, stateTree, vms, types.UnwrapSigned(candidateMsgs),
-		w.minerOwnerAddr, types.NewBlockHeight(blockHeight), ancestors)
-	if err != nil {
-		return nil, errors.Wrap(err, "generate apply messages")
-	}
+	// Dragons: ask something to select and order messages to include
```

Comment:
Good direction and I like how it removes a bunch of dependencies from the mining worker.